### PR TITLE
Stock detail page

### PR DESCRIPTION
**Pull request by alizehrehman**
- Created stock detail page component and sub-components
- Added a helper function to format data for highcharts
- Refactored to use react router (added a link to access the stock detail page component)
- Updated sampleData to be a javascript file and exported the data variable
- Added highcharts dependencies

### CODE REVIEW

File: client/src/components/StockDetailPage/Graph.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import Highcharts from 'highcharts/highstock';
+import HighchartsReact from 'highcharts-react-official';
+
+const Graph = (props) => {
```
Comment (NylonEric) [inline client/src/components/StockDetailPage/Graph.jsx:6]: This looks great!

**Review by NylonEric — CHANGES_REQUESTED** [review]
One comment about the URL route. I don't think upper case will parse in a browser, so better to go with lower and "-" dashes.

File: client/src/components/StockDetailPage.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import StockInformation from './StockDetailPage/StockInformation.jsx';
+import BuySellMenu from './StockDetailPage/BuySellMenu.jsx';
+import StockTrends from './StockDetailPage/StockTrends.jsx';
+import { sampleData } from '/sampleData/sampleData.js';
+
+class StockDetailPage extends React.Component {
```
[the remark this answers is not in the document]
Comment (alizehrehman) [reply client/src/components/StockDetailPage.jsx:7]: At the moment it's easiest to keep it stateful, but when I refactor it to use props I'll probably make it functional

File: client/src/components/App.jsx [hunk 3]
```diff
@@ -31,13 +32,17 @@ class App extends React.Component {
...
+                  <Link to="/stockDetailPage">Stock Detail Page</Link>
+                </li>
               </ul>
             </nav>
           </div>
           <Switch>
             <Route exact path="/leaderboard" component={Leaderboard} />
             <Route exact path="/trade" component={Trade} />
             <Route exact path="/login" component={Login} />
+            <Route exact path="/stockDetailPage" component={StockDetailPage} />
```
[the remark this answers is not in the document]
Comment (alizehrehman) [reply client/src/components/App.jsx:45]: Will do! I'll push another commit

**Comment by alizehrehman** [comment]
@NylonEric I pushed a new commit with the updated url. Please let me know if it looks good!

**Pull request merged**